src/include/txn_inline.h

```diff
@@ -832,6 +825,47 @@ __wt_txn_visible(
     return (timestamp <= txn_shared->read_timestamp);
 }
 
+/*
+ * __wt_txn_snap_min_visible --
+ *     Can the current transaction snapshot minimum/read timestamp see the given ID/timestamp? This
+ *     visibility check should only be used when assessing broader visibility based on aggregated
+ *     time window. It does not reflect whether a specific update is visible to a transaction.
+ */
+static inline bool
+__wt_txn_snap_min_visible(
+  WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t timestamp, wt_timestamp_t durable_timestamp)
+{
+    /* Transaction snapshot minimum check. */
+    if (!WT_TXNID_LT(id, session->txn->snap_min))
+        return (false);
+
+    /* Transactions read their writes, regardless of timestamps. */
+    if (F_ISSET(session->txn, WT_TXN_HAS_ID) && id == session->txn->id)
+        return (true);
+
+    /* Timestamp check. */
+    return (__wt_txn_timestamp_visible(session, timestamp, durable_timestamp));
+}
+
+/*
+ * __wt_txn_visible --
+ *     Can the current transaction see the given ID/timestamp?
+ */
+static inline bool
+__wt_txn_visible(
+  WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t timestamp, wt_timestamp_t durable_timestamp)
+{
+    if (!__txn_visible_id(session, id))
+        return (false);
+
+    /* Transactions read their writes, regardless of timestamps. */
+    if (F_ISSET(session->txn, WT_TXN_HAS_ID) && id == session->txn->id)
```

Comment:
```__txn_visible_id()``` already has a check:

```    /* Transactions see their own changes. */
    if (id == txn->id)
        return (true);```

So it might not be needed. But, it doesn't check for flag ```WT_TXN_HAS_ID```, which might not be needed as the id allocation and the flag are set concurrently in ```__wt_txn_id_check```

Reply:
The check in the function `__txn_visible_id()` function confirms whether this current transaction can see this update with the transaction or not.

The second check is to confirm whether we need to perform the timestamp check or not. Both checks are required. The second check already exists as part of the `__wt_txn_visible()` function before this change.

Reply:
Just checked, sounds good.